File: springboot-application/src/main/java/fr/imageupload/imageupload/model/ImageDocument.java
package fr.imageupload.imageupload.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * Classe entite representant une image stockee en base de donnees MongoDB.
 * Utilise Lombok pour generer automatiquement les getters, setters et autres
 * methodes.
 * 
 * @author Etudiant BUT 2
 * @version 1.0
 */
@Data
@Document(collection = "images")
public class ImageDocument {

    /**
     * Identifiant unique de l'image genere automatiquement par MongoDB.
     */
    @Id
    private String id;

    /**
     * Nom original du fichier image uploade par l'utilisateur.
     */
    private String name;

    /**
     * Type MIME du fichier (par exemple: image/jpeg, image/png).
     */
    private String contentType;

    /**
     * Donnees binaires de l'image stockees sous forme de tableau de bytes.
     */
    private byte[] imageData;
}

File: flask-application/models/image_document.py
# Classe représentant le modèle de données pour une image stockée dans MongoDB
class ImageDocument:
    def __init__(self, name, content_type, image_data):
        # Nom original du fichier (ex: "photo.jpg")
        self.name = name
        # Type MIME du fichier (ex: "image/jpeg")
        self.content_type = content_type
        # Données binaires du fichier (le contenu de l’image elle-même)
        self.image_data = image_data

    # Méthode pour convertir l'objet en dictionnaire utilisable par MongoDB
    def to_dict(self):
        return {
            "name": self.name,                   # Nom du fichier
            "content_type": self.content_type,   # Type MIME
            "image_data": self.image_data        # Données binaires de l'image
        }